unittest/src/fake/buttons/mod.rs

```diff
@@ -0,0 +1,126 @@
+    fn id(&self) -> u32 {
+        DRIVER_NUMBER
+    }
+    fn num_upcalls(&self) -> u32 {
+        1
+    }
+
+    fn command(&self, command_number: u32, argument0: u32, _argument1: u32) -> CommandReturn {
+        match command_number {
+            BUTTONS_COUNT => crate::command_return::success_u32(NUM_BUTTONS as u32),
+            BUTTONS_ENABLE_INTERRUPTS => {
+                if argument0 < NUM_BUTTONS as u32 {
+                    let button = self.buttons[argument0 as usize].get();
+                    self.buttons[argument0 as usize].set(ButtonState {
+                        interrupt_enabled: true,
+                        ..button
+                    });
+                    crate::command_return::success()
+                } else {
+                    crate::command_return::failure(ErrorCode::Invalid)
+                }
+            }
+            BUTTONS_DISABLE_INTERRUPTS => {
+                if argument0 < NUM_BUTTONS as u32 {
+                    let button = self.buttons[argument0 as usize].get();
+                    self.buttons[argument0 as usize].set(ButtonState {
+                        interrupt_enabled: false,
+                        ..button
+                    });
+                    crate::command_return::success()
+                } else {
+                    crate::command_return::failure(ErrorCode::Invalid)
+                }
+            }
+            BUTTONS_READ => {
+                if argument0 < NUM_BUTTONS as u32 {
+                    crate::command_return::success_u32(
+                        self.buttons[argument0 as usize].get().pressed as u32,
+                    )
+                } else {
+                    crate::command_return::failure(ErrorCode::Invalid)
+                }
+            }
+            _ => crate::command_return::failure(ErrorCode::NoSupport),
+        }
+    }
+}
+
+// -----------------------------------------------------------------------------
+// Implementation details below
+// -----------------------------------------------------------------------------
+
+#[cfg(test)]
+mod tests;
+
+const DRIVER_NUMBER: u32 = 3;
+
+// Command IDs
+const BUTTONS_COUNT: u32 = 0;
+
+const BUTTONS_ENABLE_INTERRUPTS: u32 = 1;
+const BUTTONS_DISABLE_INTERRUPTS: u32 = 2;
+
+const BUTTONS_READ: u32 = 3;
+
+impl<const NUM_BUTTONS: usize> Buttons<NUM_BUTTONS> {
+    pub fn set_pressed(&self, button: u32, pressed: bool) -> Result<(), ErrorCode> {
+        self.buttons
+            .get(button as usize)
+            .map(|button_state| {
+                let original_button_state = button_state.get();
+                button_state.set(ButtonState {
+                    pressed,
+                    ..original_button_state
+                });
+                if original_button_state.interrupt_enabled
+                    && original_button_state.pressed != pressed
+                {
+                    let _ = upcall::schedule(DRIVER_NUMBER, 0, (button, pressed as u32, 0));
```

Comment:
I don't think you want to ignore the error returned by `upcall::schedule`. Perhaps call `.expect()` instead?